conceptual/Npgsql/release-notes/6.0.md

```diff
@@ -0,0 +1,166 @@
+
+## Breaking changes
+
+### Major changes to timestamp mapping
+
+> [!NOTE]
+> It is possible to opt out of these changes to maintain backwards compatibility, see below.
+
+Note: The below notes will use the PostgreSQL aliases `timestamptz` to refer to `timestamp with time zone`, and `timestamp` to refer to `timestamp without time zone`. Note that `timestamp with time zone` represents a UTC timestamp and does **not** store a timezone in the database.
+
+* UTC DateTime is now strictly mapped to `timestamptz`, while Local/Unspecified DateTime is now strictly mapped to `timestamp`.
+  * DateTime with Kind=UTC are now written as PostgreSQL `timestamptz`; previously, DateTime was always written as `timestamp`.
+  * It is no longer possible to write UTC DateTime as `timestamp`, or Local/Unspecified DateTime as `timestamptz`. This was previously allowed, with Npgsql performing implicit timezone conversions.
+  * Note that if you write a UTC DateTime to a PostgreSQL `timestamp` column, PostgreSQL will implicitly convert the `timestamptz` value sent by Npgsql, performing a timezone conversion based on the `TimeZone` parameter.
+* `timestamptz` values are now read back as DateTime with Kind=UTC, without any conversions; these were previously returned as local DateTime, converted to the local machine's timezone. When reading `timestamptz` values as [DateTimeOffset](https://docs.microsoft.com/dotnet/api/system.datetimeoffset), UTC values (offset 0) are always returned.
+* It is no longer possible to write [DateTimeOffset](https://docs.microsoft.com/dotnet/api/system.datetimeoffset) with offsets other than 0 (UTC), since these cannot be represented in PostgreSQL. These were previously implicitly converted to UTC before sending.
+* It is no longer possible to read or write `timetz` as DateTime or TimeSpan, as these don't have a timezone. This was previously allowed, with the offset being stripped.
+
+See the [EF Core provider docs](/efcore/release-notes/6.0.html) for additional changes at the EF level.
+
+#### NodaTime changes
+
+* [Instant](https://nodatime.org/3.0.x/api/NodaTime.Instant.html) is now sent as a `timestamptz` value, and not as a `timestamp`, since they represent a universally agreed-upon point in time. To send a `timestamp`, use [LocalDateTime](https://nodatime.org/3.0.x/api/NodaTime.LocalDateTime.html).
+* When reading `timestamptz` as [ZonedDateTime](https://nodatime.org/3.0.x/api/NodaTime.ZonedDateTime.html) or [OffsetDateTime](https://nodatime.org/3.0.x/api/NodaTime.OffsetDateTime.html), UTC values are always returned. Previously, local values based on the PostgreSQL `TimeZone` parameter were returned.
```

Comment:
Pg timezone or machine timezone? Was there really a difference between nodatime and bcl handling?

Reply:
Yeah, there actually was... The BCL types could only work with Windows timezones (until .NET 6.0 that is), whereas NodaTime has always supported IANA/Olson, so we actually looked at TimeZone in an effort to be even more conforming to the PG behavior... All seems pretty senseless now.

Reply:
TIL :o

Reply:
We even had https://github.com/npgsql/npgsql/issues/3594 :rofl: